doc: build: dts: Fix incorrect foreach documentation descriptions

rruuaanng opened the pull request:
Replaced incorrect rst link references in the description of traversing ancestor nodes, along with some conflicting descriptions.

rruuaanng commented:
Hmm... It seems that I did not modify the corresponding pointer after copying the rst tag :(

nordicjm commented on doc/build/dts/api/api.rst:
```diff
@@ -104,12 +104,10 @@ does not apply to macros which take cell names as arguments.
-There is currently only one "generic" for-each macro,
-:c:macro:`DT_FOREACH_CHILD`, which allows iterating over the children of a
-devicetree node.
+Additionally, it provides another macro :c:macro:`DT_FOREACH_CHILD`, which allows
```
```suggestion
Additionally, the :c:macro:`DT_FOREACH_CHILD` macro allows
```
(adjust line length)

rruuaanng replied:
Done.

rruuaanng committed:
doc: build: dts: Fix incorrect foreach documentation

Replaced incorrect rst link references in the description
of traversing ancestor nodes, along with some conflicting
descriptions.

Signed-off-by: James Roy <rruuaanng@outlook.com>
